### PR TITLE
kube-proxy: windows: Removed hnsV1

### PR DESCRIPTION
**Pull request by papagalu**
hnsV1 not supported anymore

Signed-off-by: Dimitrie Mititelu <dmititelu@cloudbasesolutions.com>

#### What type of PR is this?
/kind cleanup

#### Does this PR introduce a user-facing change?
```release-note
Windows winkernel Kube-proxy no longer supports Windows HNS v1 APIs
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by jsturtevant** [comment]
/sig windows
/release-note Windows winkernel Kube-proxy no longer supports Windows HNS v1 APIs
/triage accepted
/ok-to-test
/priority important-soon
/milestone v1.25

File: pkg/proxy/winkernel/proxier.go [hunk 1]
```diff
@@ -146,14 +146,15 @@ type remoteSubnetInfo struct {
...
 func newHostNetworkService() (HostNetworkService, hcn.SupportedFeatures) {
-	var hns HostNetworkService
-	hns = hnsV1{}
+	var h HostNetworkService
 	supportedFeatures := hcn.GetSupportedFeatures()
 	if supportedFeatures.Api.V2 {
-		hns = hnsV2{}
+		h = hns{}
+	} else {
+		panic("Api V2 not supported")
```
Comment (jsturtevant) [inline pkg/proxy/winkernel/proxier.go:154]:
```suggestion
		panic("Windows HNS Api V2 required.  This version of windows does not support API V2")
```

maybe something like that?

**Comment by jsturtevant** [comment]
/cc @daschott @sbangari

@papagalu could you add a release note?  I tried to via the bot but doesn't look like it worked

**Comment by harshitasao** [comment]
Hi @papagalu, k8s 1.25 Bug Triage Shadow here, just checking in to see if this is still on track for k8s 1.25?
Thank you.

**Review by marosset — APPROVED** [review]

**Comment by AbelHu** [comment]
@marosset May I know which component relies on hnsv1 api, docker or WS2019?

**Comment by jsturtevant** [comment]
dockershim uses the v1 HNS api, We've actually already broken support for this via https://github.com/kubernetes/kubernetes/pull/109124. 
 See https://github.com/kubernetes/kubernetes/issues/110591

**Comment by daschott** [comment]
Yes, WS2019 supports both v2 and v1. Duplicating code for both v1/v2 is a big maintenance overhead so we should use only v2 going forward. v1 is used on Windows Server versions before Windows Server 2019. However since these older OS versions are not supported by Kubernetes this should be safe to remove.

**Comment by marosset** [comment]
HNS apis can also be used directly by CNI binaries/plugins.
I think @daschott looked at the more common CNI plugins (flannel, calico, azure-cni, antrea...) and concluded that they are all supported v2 HNS apis.

**Comment by AbelHu** [comment]
Thank all of you for the answers. It seems like that we do not need to do anything with this change since we are using WS2019/WS2022 and docker will be deprecated in k8s 1.24+.

**Review by feiskyer — APPROVED** [review]
lgtm

**Comment by daschott** [comment]
These changes LGTM but reviewing this there are still some v1 calls left in proxier.go (`hcsshim.<function>`) that should probably also be substituted with v2 equivalents `(hcn.<function>)` as another PR.

**Pull request merged**